### PR TITLE
Weighted NFTs

### PR DESCRIPTION
**Pull request by tenthirtyone**
## Preamble

eip: 874
title: Weighted NFTs
author: Alex Sherbuck <Alex@igave.io>
type: Standard Track
category: ERC
status: Draft
created: 2018-02-06

##  Summary
Not all NFTs are created equal. Nothing exists to distinguish one NFT from another. Weight allows a DAO or other external actors to recognize some value for unique asset holdings.

## Motivation
This allows DAOs to form based on asset/NFT holdings instead of just ERC-20 token holdings. Asset holders may decide how to set their token weight.

## Specification
The Weighted Digital Asset Registry is an extension for ERC-821 that tracks asset weight. This follows the standard set in [ERC821](https://github.com/ethereum/EIPs/issues/821) and adds:

## Weighted DAR
Extension to ERC-821

#### totalWeight
`function totalWeight() public view returns (uint256);`
Returns the total weight tracked by the DAR

#### isWeighted
`function isWeighted() public view returns (bool);`
This method returns true.

#### weightOfAsset
`function weightOfAsset(uint256 assetId) public view returns (uint64);`
Returns the total weight of an individual asset.

#### weightOfHolder
`function weightOfHolder(address holder) public view returns (uint256);`
Returns the total weight of the assets controlled by the holder.

#### changeWeight
`function changeWeight(uint256 assetId, uint64 weight) public;`
Changes the weight of the asset to the given value.

#### Events
```
event TransferWeight(
    address indexed from,
    address indexed to,
    uint256 indexed assetId,
    uint64 weight
  );
event ChangeWeight(
    uint256 indexed assetId,
    uint64 weight
  );
```

## DAR Adapter
An adapter that exposes a `balanceOf`. 

#### voteWeightAddress
`function voteWeightAddress() public view returns (address);`
Returns the address of the Weighted DAR.

#### balanceOf
`function balanceOf(address holder) public view returns  (uint256);`
Calls the `weightOfHolder` function of the Weighted DAR.

## Implementation
This will be updated to the most recent ERC-721 standard. Update: Standard is settled. Getting to work

[Weighted Digital Asset Registry](https://github.com/I-Gave/erc821/tree/weighted-registry/contracts) - w/ DAO voting

## Rationale
Including an adapter for the DAO to reference holder weight minimizes the impact on current DAO code that references an ERC-20's `balanceOf`

## Copyright
Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).

### CODE REVIEW

File: EIPS/DAR-vote-weight.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    Author: Alex Sherbuck <Alex@igave.io>
+    Type: Standard Track
+    Category: ERC
+    Status: Draft
+    Created: 2018-02-06
+    Requires EIP 821
+
+
+##  Summary
+The ERC-721 NFT `balanceOf` returns the count of assets owned by an address. ERC-821 moved this functionality into `assetCount` and maintains `balanceOf` as an alias to `assetCount`. This change proposes `balanceOf` return an abstract weight representing the NFTs value in the system.
```
Comment (pirapira) [inline EIPS/DAR-vote-weight.md]: Maybe "NFT's" instead of "NFTs"?

File: EIPS/DAR-vote-weight.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+## Motivation
+The ERC-20 `balanceOf` works well for DAO voting. A fungible tokens value corresponds 1:1 with its asset count. The NFT `balanceOf` should represent the tokens value.
+
+E.g. Each Cryptokitty has a generation. Each plot of LAND has an auction price. An NFT is like a receipt for that value. 'Weight' is the abstraction of that property.
+
+Voting based on asset weight permits DAOs to form around NFC voting or a combination of ERC-20 and NFC voting. Instead of associations of token holders this permits associations of LAND holders, for instance.
+
+## Specification
+
+An ERC-721 example exists but the recent ERC-821 DAR is better suited for this extension. The ERC-721 example is included at the end, the ERC-821 rework is taking place. This spec will be updated to reflect the completion of that work.
```
Comment (pirapira) [inline EIPS/DAR-vote-weight.md]: Please specify something.

Comment (tenthirtyone) [reply EIPS/DAR-vote-weight.md]: @pirapira updated

**Comment by Arachnid** [comment]
This is a courtesy notice to let you know that the format for EIPs has been modified slightly. If you want your draft merged, you will need to make some small changes to how your EIP is formatted:

 - Frontmatter is now contained between lines with only a triple dash ('---')
 - Headers in the frontmatter are now lowercase.

If your PR is editing an existing EIP rather than creating a new one, this has already been done for you, and you need only rebase your PR.

In addition, a continuous build has been setup, which will check your PR against the rules for EIP formatting automatically once you update your PR. This build ensures all required headers are present, as well as performing a number of other checks.

Please rebase your PR against the latest master, and edit your PR to use the above format for frontmatter. For convenience, here's a sample header you can copy and adapt:

```
---
eip: <num>
title: <title>
author: <author>
type: [Standards Track|Informational|Meta]
category: [Core|Networking|Interface|ERC] (for type: Standards Track only)
status: Draft
created: <date>
---
```

File: EIPS/DAR-vote-weight.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+## Specification
+The Weighted Digital Asset Registry is an extension for ERC-821 that tracks asset weight. This follows the standard set in [ERC821](https://github.com/ethereum/EIPs/issues/821) and adds:
+
+## Weighted DAR
+Extension to ERC-821
+
+#### totalWeight
+`function totalWeight() public view returns (uint256);`
+Returns the total weight tracked by the DAR
```
Comment (axic) [inline EIPS/DAR-vote-weight.md]: What is a DAR?

**Comment by axic** [comment]
@tenthirtyone are you still interested in pursuing this ERC?

**Pull request closed without merging**